### ISSUE TITLE
Implement lander MVP

### ISSUE THREAD

**Issue by taneliang** [taneliang/stack-attack#39]
## Requirements

- Support branch status check and up to date requirements
- Minimize CI time
- Partial stack landing

## Out of scope

- Atomic (but still possibly partial) stack landing -- use feature flags to
  prevent breaking master

## Interface

Commands

- `/land` or `@sttack land`: land this PR and all the PRs its based on
- `/rebase` or `@sttack restack` or `@sttack rebase`: rebase stack below this PR onto the updated base stack or on the
  updated master

### Landing

- `/land` received
- Find all PRs in stack
- For each unmerged open PRs in stack, in order of dependency:
  - If PR is outdated:
    - If PR is not a Stack Attack PR, abort landing.
    - Rebase the PR's one commit onto its base branch. If cannot rebase, abort landing.
  - If status checks required, wait for required status checks to pass.
  - Squash or rebase merge (actually just whatever they like, let's default to squash if we can't find out the repo default).
  - If cannot merge PR, abort with comment

## Implementation plan

- [x] Set up basic repository (https://github.com/taneliang/stack-attack-bot)
- [x] Set up logging/monitoring codebase infra
- [x] Decide on basic codebase organization
- [x] Implement PR comment command detection
- [x] Implement rebase of 1 PR onto its base branch
- [ ] ~Implement landing of 1 PR onto its base branch~
- [x] Implement stack detection within a PR
- [ ] ~Implement status check change detection (to trigger PR landing)~
- [ ] ~Implement sequential PR landing (store to-be-landed state in a PR comment)~
- [ ] ~Implement sequential PR rebase~
- [x] Set up deployment (on Heroku?) (on Heroku.)
- [x] Sentry
- [x] Logentries
- [x] CI/CD

## Post-MVP tasks

Gathering a laundry list of things that can be done. To be split out into individual issues and further prioritized once the MVP is complete.

- Staging bot
- E2E tests?

**Comment by taneliang** [comment]
Descoped to ship. Implemented a bot that only rebases single PRs, which provides the largest UX improvement without incurring the complexity of landing (which has security concerns) and the complexity of applying an operation sequentially on a stack.

- App page: https://github.com/apps/stack-attack-bot
- Repo: https://github.com/StackAttack/stack-attack-bot
